From 69b71356b74faee074136ca05c78e0bed72f13d8 Mon Sep 17 00:00:00 2001
From: Ronny Lorenz <ronny@tbi.univie.ac.at>
Date: Sun, 16 Apr 2017 21:11:56 +0200
Subject: [PATCH 4/4] Fix rk3288.dtsi and add Tinkerboard changes to
 rk3288-miniarm.dts

This applies changes made available through Armbian and some more upstream fixes
---
 arch/arm/boot/dts/rk3288-miniarm.dts | 99 +++++++++++++++++++++++++++++++-----
 arch/arm/boot/dts/rk3288.dtsi        |  8 +--
 2 files changed, 90 insertions(+), 17 deletions(-)

diff --git a/arch/arm/boot/dts/rk3288-miniarm.dts b/arch/arm/boot/dts/rk3288-miniarm.dts
index 609e8588795a..5fe4547bf52c 100644
--- a/arch/arm/boot/dts/rk3288-miniarm.dts
+++ b/arch/arm/boot/dts/rk3288-miniarm.dts
@@ -42,10 +42,24 @@
 
 #include <dt-bindings/clock/rockchip,rk808.h>
 #include "rk3288.dtsi"
+#include <dt-bindings/input/input.h>
 
 / {
 	compatible = "rockchip,rk3288-miniarm", "rockchip,rk3288";
 
+	ion {
+		compatible = "rockchip,ion";
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		cma-heap {
+			reg = <0x78000000 0x08000000>;
+		};
+
+		system-heap {
+		};
+	};
+
 	memory {
 		device_type = "memory";
 		reg = <0x0 0x80000000>;
@@ -66,7 +80,7 @@
 	wireless-wlan {
 		compatible = "wlan-platdata";
 		rockchip,grf = <&grf>;
-		wifi_chip_type = "ap6212";
+		wifi_chip_type = "rtl8723bs";
 		sdio_vref = <1800>;
 		WIFI,host_wake_irq = <&gpio4 30 GPIO_ACTIVE_HIGH>;
 		status = "okay";
@@ -79,12 +93,6 @@
 		#clock-cells = <0>;
 	};
 
-	io-domains {
-		compatible = "rockchip,rk3288-io-voltage-domain";
-		rockchip,grf = <&grf>;
-		wifi-supply = <&vcc_18>;
-	};
-
 	sdio_pwrseq: sdio-pwrseq {
 		compatible = "mmc-pwrseq-simple";
 		clocks = <&rk808 RK808_CLKOUT1>;
@@ -129,8 +137,13 @@
 		};
 
 		act-led {
-			gpios=<&gpio2 3 GPIO_ACTIVE_LOW>;
-			linux,default-trigger="mmc0";
+			gpios = <&gpio1 0x18 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "mmc0";
+		};
+		
+		heartbeat-led {
+			gpios = <&gpio1 0x19 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "heartbeat";
 		};
 	};
 
@@ -172,6 +185,13 @@
 		startup-delay-us = <100000>;
 		vin-supply = <&vcc_io>;
 	};
+
+	vdd_log: vdd_log {
+		compatible = "regulator-fixed";
+		regulator-min-microvolt = <1000000>;
+		regulator-max-microvolt = <1000000>;
+		regulator-name = "vdd_logic";
+	};
 };
 
 &cpu0 {
@@ -191,7 +211,7 @@
 	pinctrl-0 = <&rgmii_pins>;
 	tx_delay = <0x30>;
 	rx_delay = <0x10>;
-	status = "ok";
+	status = "okay";
 };
 
 &gpu {
@@ -413,6 +433,12 @@
 
 &i2c2 {
 	status = "okay";
+
+		eeprom:m24c08@50 {
+            compatible = "at,24c08";
+            reg = <0x50>;
+        };
+
 };
 
 &i2s {
@@ -421,9 +447,9 @@
 };
 
 &io_domains {
-	status = "okay";
-
+	wifi-supply = <&vcc_18>;
 	sdcard-supply = <&vccio_sd>;
+	status = "okay";
 };
 
 &sdio0 {
@@ -439,7 +465,7 @@
 	non-removable;
 	num-slots = <1>;
 	pinctrl-names = "default";
-	pinctrl-0 = <&sdio0_bus4 &sdio0_cmd &sdio0_clk>;
+	pinctrl-0 = <&sdio0_bus4 &sdio0_cmd &sdio0_clk &sdio0_int>;
 	sd-uhs-sdr104;
 	supports-sdio;
 };
@@ -461,6 +487,10 @@
 	bus-width = <4>;
 	cap-mmc-highspeed;
 	cap-sd-highspeed;
+	sd-uhs-sdr12;
+	sd-uhs-sdr25;
+	sd-uhs-sdr50;
+	sd-uhs-sdr104;
 	card-detect-delay = <200>;
 	disable-wp;			/* wp not hooked up */
 	num-slots = <1>;
@@ -479,6 +509,8 @@
 };
 
 &uart0 {
+	dmas = <&dmac_peri 1>, <&dmac_peri 2>;
+	dma-names = "tx", "rx";
 	pinctrl-names = "default";
 	pinctrl-0 = <&uart0_xfer>, <&uart0_cts>;
 	status = "okay";
@@ -573,6 +605,12 @@
 		};
 	};
 
+	sdio-pwrseq {
+		wifi_enable_h: wifi-enable-h {
+			rockchip,pins = <4 28 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+	};
+
 	pmic {
 		pmic_int: pmic-int {
 			rockchip,pins = <RK_GPIO0 4 RK_FUNC_GPIO &pcfg_pull_up>;
@@ -634,4 +672,39 @@
 		};
 	};
 
+	cam_pins {
+		cam0_default_pins:cam0_default_pins {
+			rockchip,pins = <0 17 RK_FUNC_GPIO &pcfg_pull_none>,/*sensor power pin: GPIO0_C1 */
+			<2 0 RK_FUNC_GPIO &pcfg_pull_none>,/*sensor power down pin: GPIO2_A0*/
+			<2 11 RK_FUNC_1 &pcfg_pull_none>;/*sensor mclk: cif_clkout*/
+		};
+
+		cam0_sleep_pins:cam0_sleep_pins {
+			rockchip,pins = <0 17 RK_FUNC_GPIO &pcfg_pull_none>,/*sensor power pin: GPIO0_C1 */
+			<2 0 RK_FUNC_GPIO &pcfg_pull_none>,/* cif power down pin */
+			<2 11 RK_FUNC_GPIO &pcfg_pull_none>;/* cif_clkout */
+		};
+	};
+};
+
+&dmc {
+	status = "okay";
+	logic-supply = <&vdd_log>;
+	rockchip,odt-disable-freq = <666000000>;
+	rockchip,dll-disable-freq = <333000000>;
+	rockchip,sr-enable-freq = <333000000>;
+	rockchip,auto-self-refresh-cnt = <0>;
+	rockchip,auto-power-down-cnt = <64>;
+	rockchip,ddr-speed-bin = <21>;
+	rockchip,trcd = <10>;
+	rockchip,trp = <10>;
+	rockchip,pd-enable-freq = <333000000>;
+	operating-points = <
+		/* KHz    uV */
+		200000 1000000
+		333000 1000000
+		533000 1000000
+	>;
+
+	#cooling-cells = <2>; /* min followed by max */
 };
diff --git a/arch/arm/boot/dts/rk3288.dtsi b/arch/arm/boot/dts/rk3288.dtsi
index 1a8de87b02d1..532d98977ef7 100644
--- a/arch/arm/boot/dts/rk3288.dtsi
+++ b/arch/arm/boot/dts/rk3288.dtsi
@@ -1043,7 +1043,7 @@
 		status = "disabled";
 	};
 
-	spdif: sound@ff88b0000 {
+	spdif: sound@ff8b0000 {
 		compatible = "rockchip,rk3288-spdif", "rockchip,rk3066-spdif";
 		reg = <0xff8b0000 0x10000>;
 		#sound-dai-cells = <0>;
@@ -1213,7 +1213,7 @@
 	mipi_dsi: mipi@ff960000 {
 		compatible = "rockchip,rk3288-mipi-dsi", "snps,dw-mipi-dsi";
 		reg = <0xff960000 0x4000>;
-		interrupts = <GIC_SPI 83 IRQ_TYPE_LEVEL_HIGH>;
+		interrupts = <GIC_SPI 19 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&cru SCLK_MIPIDSI_24M>, <&cru PCLK_MIPI_DSI0>;
 		clock-names = "ref", "pclk";
 		power-domains = <&power RK3288_PD_VIO>;
@@ -1506,7 +1506,7 @@
 		#address-cells = <0>;
 
 		reg = <0xffc01000 0x1000>,
-		      <0xffc02000 0x1000>,
+		      <0xffc02000 0x2000>,
 		      <0xffc04000 0x2000>,
 		      <0xffc06000 0x2000>;
 		interrupts = <GIC_PPI 9 0xf04>;
@@ -1756,7 +1756,7 @@
 				rockchip,pins = <6 21 RK_FUNC_1 &pcfg_pull_up>;
 			};
 
-			sdmmc_cd: sdmcc-cd {
+			sdmmc_cd: sdmmc-cd {
 				rockchip,pins = <6 22 RK_FUNC_1 &pcfg_pull_up>;
 			};
 
-- 
2.12.2

